styles/lib/carnival/features/_tables.scss

```diff
@@ -0,0 +1,355 @@
+      }
+      & .os-caption-container {
+        display: table-caption;
+        caption-side: bottom;
+        prince-caption-page: all;
+        font-size: 0.8333333333rem;
+        line-height: 1.2;
+        > .os-title-label {
+          font-family: fonts.$mulish;
+          font-weight: bold;
+          color: colors.$R1;
+          text-transform: uppercase;
+        }
+        > .os-number {
+          font-family: fonts.$mulish;
+          font-weight: bold;
+          color: colors.$R1;
+        }
+        > .os-title {
+          font-family: fonts.$mulish;
+          font-weight: bold;
+        }
+        > .os-caption {
+          font-family: fonts.$plex;
+        }
+      }
+    }
+}
+
+@mixin column-header-table($book-table-color: colors.$black){
+  .os-table.os-column-header-container {
+    margin-bottom: 0.7rem;
+    display: table;
+    margin-left: auto;
+    margin-right: auto;
+    > table {
+      border-collapse: collapse;
+      > tbody
+        > tr {
+          > td{
+            &:not(:only-of-type) {
+              font-family: fonts.$plex;
+              border-left: 0.025rem solid colors.$black;
+              border-bottom: 0.025rem solid colors.$black;
+              padding-top: 0.7rem;
+              padding-bottom: 0.7rem;
+              padding-right: 0.7rem;
+              padding-left: 0.7rem;
+            }
+            &:not(:only-of-type):not(:last-child) {
+              border-left: none;
+              border-right: 0.1rem solid colors.$black;
+              font-family: fonts.$mulish;
+              font-weight: 700;
+            }
+          }
+        &:first-of-type {
+          border-top-color: $book-table-color;
+          border-top-width: 0.1rem;
+          border-top-style: solid;
+        }
+        &:last-of-type > td {
+          border-bottom-color: $book-table-color;
+          border-bottom-width: 0.1rem;
+          border-bottom-style: solid;
+        }
+      }
+    }
+  }
+}
+
+@mixin vertically-tight-table(){
+  .os-table.os-vertically-tight-container > table.vertically-tight {
+    > thead > tr > th {
+      padding-top: 0.16rem;
+      padding-bottom: 0.16rem;
+    }
+    > tbody > tr > td:not(.regular-spacing):not(.pseudo-class-for-precedence) {
+      padding-top: 0.16rem;
+      padding-bottom: 0.16rem;
+    }
+  }
+}
+
+@mixin table-from-exercise(){
+  .os-eoc {
+    & [data-type=problem], [data-type=exercise-question] {
+      > div .os-table > table {
+        max-width: 100%;
+        table-layout: fixed;
+        > tbody > tr > td:not(:only-of-type) {
+          text-align: center;
+          padding-top: 0.7rem;
+          padding-bottom: 0.7rem;
+          padding-right: 4px;
+          padding-left: 4px;
+        }
+      }
+    }
+    > .os-problem-container, .os-solution-container {
+      > p + .os-table {
+        margin-top: 1rem;
+      }
+    }
+  }
+  .os-eob {
+    > .os-solution-container {
+      > p + .os-table {
+        margin-top: 1rem;
+      }
```

Comment:
i think this would still be clearer as 
```
.os-eoc .os-problem-container, .os-eob .os-solution-container {
  > p + .os-table {
    margin-top: 1rem;
  }
}
```

because it groups the properties on the selector. the selector `.os-eoc .os-solution-container` shouldn't be needed because `solutions` are usually moved to `eob`